src/helpers/time-to-first-byte-analyser.helper.spec.js

```diff
@@ -0,0 +1,27 @@
+import { create as createAnalyser } from './time-to-first-byte-analyser.helper';
+import { expect } from 'chai';
+
+const myFakeParser = {
+  parse: () => {
+    return [{ttfb: 0, url: 'http://localhost:8080/'},
+      {ttfb: 2, url: 'http://localhost:8080/assets/kittens.jpg'},
+      {ttfb: 1, url: 'http://localhost:8080/favicon.ico'},
+      {ttfb: 2, url: 'http://localhost:8080/tabular-data'},
+      {ttfb: 1, url: 'http://localhost:8080/assets/kittens.jpg'}]
+  }
+};
+const analyser = createAnalyser(myFakeParser);
+
+describe('Time to first byte analyser', () => {
+  it('returns found slow request', () => {
+    const maxTiming = 0; // equal to the lowest request's TTFB value
+
+    expect(analyser.checkTiming(myFakeParser, maxTiming)).to.deep.include({ttfb: 2, url: 'http://localhost:8080/assets/kittens.jpg'});
```

Comment:
Why should it return this value? I don't get it - maxTiming = 0, so all requests should have failed.

Reply:
It doesn't return just this value.

I am testing here, if a correct value has been returned and if it contains expect `slow request`.
http://chaijs.com/api/bdd/

Reply:
Move the array you have in `myFakeParser` to a variable, and then instead of checking a single item here, just do the `to.equal(theWholeArrayOfRequests` this way you'll be sure that it filtered correct requests.

What's more it would be better to test actual filtering, so change maxTiming to 1 and check if the result of method has a expected length and items.

Reply:
Yeah, now I see theres `.include` part.
Adam is correct - it's always better to test te more hard parts, not only the _happy paths_. Especially in unit tests ;)

Reply:
You're right 👍